File: ProjectPos.BackEnd/ProjectPos.Services/DTOs/AccountCategoryDto.cs
using ProjectPos.Data.EntityModels;
using ProjectPos.Data.Shared.Enums;
using ProjectPos.Services.EntityDtos;

namespace ProjectPos.Services.DTOs
{
    public class AccountCategoryDto : AuditedEntityDto<int>
    {
        public string? Name { get; set; }
        public string? Description { get; set; }
        public AccountType? AccountType { get; set; }

        public List<AccountDto>? Accounts { get; set; }
    }
}

File: ProjectPos.BackOffice.FrontEnd/src/proxy/interfaces/account-category-dto.ts
import { AuditedEntityDto } from "../entity-dtos/audited-entity-dto";
import { AccountType } from "../enums/account-type";

export interface AccountCategoryDto extends AuditedEntityDto<number> {
    name?: string; // Optional string property
    description?: string; // Optional string property
    accountType?: AccountType;
}
